src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -833,6 +859,29 @@ class PersonAgent(
           } else {
             currentVehicle
           }
+        // todo rrp
+        currentVehicleForNextState.headOption.foreach { beamVehicleId =>
+          import beamScenario.beamConfig.beam.agentsim.agents.vehicles.{
+            noRechargeThresholdInMeters,
+            rechargeRequiredThresholdInMeters
+          }
+
+          val vehicle = beamVehicles(beamVehicleId).vehicle
+          if (vehicle.isBEV || vehicle.isPHEV) {
+            val distanceToDestinationInMeters = trip.foldLeft(0.0)(_ + _.beamLeg.travelPath.distanceInM)
+            val newRechargeRequiredThresholdInMeters = {
+              rechargeRequiredThresholdInMeters +
+              (1 - rechargeRequiredThresholdInMeters / distanceToDestinationInMeters) *
+              distanceToDestinationInMeters
+            }
+            // todo: compute new noRechargeThresholdInMeters using distanceToDestinationInMeters value.
+            val newNoRechargeThresholdInMeters = noRechargeThresholdInMeters
```

Comment:
@haitamlaarabi, Could you please provide equation for `noRechargeThresholdInMeters` as well? I can't figure it out how do that.